xdsl/frontend/const.py

```diff
@@ -0,0 +1,35 @@
+import ast
+from dataclasses import dataclass
+from typing import Generic, TypeVar
+
+_T = TypeVar("_T")
+
+
+class Const(Generic[_T]):
+    """
+    Marks the value as a compile-time constant and can be used in any frontend.
+    Compile-time constants can be expressions and are evaluated at compile time.
+    They cannot be redefined in the same program but can be shadowed by .
```

Comment:
What type of values are actually allowed in `Const`?
Is it only `i64`, or also other values depending if the frontend is extended?
If it is only certain values, I would document it here.

Reply:
Any values, actually. Let me add this to the documentation then.